extras/indexing/src/main/java/mvm/rya/sail/config/RyaSailFactory.java

```diff
@@ -27,73 +19,132 @@
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.Instance;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.hadoop.conf.Configuration;
+import org.openrdf.sail.Sail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
 
 import mvm.rya.accumulo.AccumuloRdfConfiguration;
 import mvm.rya.accumulo.AccumuloRyaDAO;
+import mvm.rya.accumulo.instance.AccumuloRyaInstanceDetailsRepository;
 import mvm.rya.api.RdfCloudTripleStoreConfiguration;
+import mvm.rya.api.instance.RyaDetailsRepository.RyaDetailsRepositoryException;
+import mvm.rya.api.instance.RyaDetailsToConfiguration;
 import mvm.rya.api.persist.RyaDAO;
 import mvm.rya.api.persist.RyaDAOException;
 import mvm.rya.indexing.accumulo.ConfigUtils;
 import mvm.rya.mongodb.MongoDBRdfConfiguration;
 import mvm.rya.mongodb.MongoDBRyaDAO;
+import mvm.rya.mongodb.instance.MongoRyaInstanceDetailsRepository;
 import mvm.rya.rdftriplestore.RdfCloudTripleStore;
 import mvm.rya.rdftriplestore.inference.InferenceEngine;
 import mvm.rya.rdftriplestore.inference.InferenceEngineException;
 
 public class RyaSailFactory {
-
-
+    private static final Logger LOG = LoggerFactory.getLogger(RyaSailFactory.class);
 
     public static Sail getInstance(final Configuration conf) throws AccumuloException,
-    AccumuloSecurityException, RyaDAOException, InferenceEngineException, NumberFormatException, UnknownHostException {
-
+        AccumuloSecurityException, RyaDAOException, InferenceEngineException {
         return getRyaSail(conf);
     }
 
-
-
-    private static Sail getRyaSail(final Configuration config) throws AccumuloException, AccumuloSecurityException, RyaDAOException, InferenceEngineException, NumberFormatException, UnknownHostException {
-
+    private static Sail getRyaSail(final Configuration config) throws InferenceEngineException, RyaDAOException, AccumuloException, AccumuloSecurityException {
         final RdfCloudTripleStore store = new RdfCloudTripleStore();
-        RyaDAO crdfdao = null;
-        RdfCloudTripleStoreConfiguration conf;
-        if (ConfigUtils.getUseMongo(config)) {
-            conf = new MongoDBRdfConfiguration(config);
-            conf.setTablePrefix(config.get(RdfCloudTripleStoreConfiguration.CONF_TBL_PREFIX));
-            ConfigUtils.setIndexers(conf);
-
-            crdfdao = new MongoDBRyaDAO((MongoDBRdfConfiguration)conf);
-            crdfdao.init();
-
-            conf.setDisplayQueryPlan(true);
-            store.setRyaDAO(crdfdao);
+        final RyaDAO dao;
+        final RdfCloudTripleStoreConfiguration rdfConfig;
+
+        final String user = config.get(ConfigUtils.CLOUDBASE_USER);
+        final String pswd = config.get(ConfigUtils.CLOUDBASE_PASSWORD);
```

Comment:
Could we make sure the "CLOUDBASE" username and password are also meant to be used for Mongo? That isn't very intuitive.